### ISSUE TITLE
Head managment, dynamic page title

### ISSUE THREAD

**Issue by pi0neer** [vuejs/vue-ssr-docs#304]
Apparantly, current documentation is wrong. I'm getting undefined error, when i'm trying to get item from store and put property of this item to title. I'm using latest version of this package - 2.6.12.

**Comment by claudiodekker** [comment]
Yeah, the problem is indeed that there's no specification of how this should work, and that flow-wise, the method order seems a bit backwards.

Here's an example:
```js
createRenderer({
  template: (result, context) => {
    return `<html>
      <head>${context.head}</head>
      <body>${result}</body>
    <html>`
  }
})
    .renderToString(app, {})
```

Note that empty object passed as the 2nd argument of `renderToString`. 
By default, this object is basically `undefined`.

The way that this flow actually seems to work, is like this:
1. Call to `renderToString` occurs
2. `ssrContext` gets set internally
3. Dynamic page title mixin sets a property on the now-set `ssrContext`
4. The app finishes rendering
5. The `template` callback above gets called, with `result` containing your string-rendered app, and `context` containing the input context as well as the modifications made inside the app.

Furthermore, [the example that OP mentioned](https://ssr.vuejs.org/guide/head.html) here does vaguely mention you need to 'fix' the example by configuring a plugin. Instead, I'd suggest just modifying the last few lines to use the `global` state, as that's way easier and actually fixes it while debugging:

```js
// `VUE_ENV` can be injected with `webpack.DefinePlugin`
export default global && global.process.env.VUE_ENV === 'server'
    ? serverTitleMixin
    : clientTitleMixin
```

Hope this helps someone else!

Related:
- https://github.com/vuejs/vue/issues/5941
- https://github.com/vuejs/vue/pull/5981
